clients/src/main/java/org/apache/kafka/common/metrics/JmxReporter.java

```diff
@@ -61,17 +61,17 @@
 
     private static final Logger log = LoggerFactory.getLogger(JmxReporter.class);
     private static final Object LOCK = new Object();
-    private String prefix;
+    private String prefix = "";
     private final Map<String, KafkaMbean> mbeans = new HashMap<>();
     private Predicate<String> mbeanPredicate = s -> true;
 
     public JmxReporter() {
-        this("");
```

Comment:
I think it's fine to keep calling the deprecated constructor to ensure we only have one code path to construct the reporter.